source/Irrlicht/COSOperator.cpp

```diff
@@ -26,6 +29,19 @@
 
 #include "fast_atof.h"
 
+#if defined(_IRR_COMPILE_WITH_SDL_DEVICE_)
+static bool sdl_supports_primary_selection = [] {
+#if SDL_VERSION_ATLEAST(2, 25, 0)
+	SDL_version linked_version;
+	SDL_GetVersion(&linked_version);
+	return (linked_version.major == 2 && linked_version.minor >= 25)
+			|| linked_version.major > 2;

```

Comment:
this won't work.
by referencing the `SDL_SetPrimarySelectionText` symbol you already introduce a hard-dependency

Reply:
I've put `SDL_SetPrimarySelectionText` into an `#if` (see below), so it is only referenced if the compiled SDL version is new enough.

Reply:
Yes, but the check I'm commenting on _doesn't do anything_.
If you compile with an old SDL, the code is never included.
If you compile with a new SDL, the code is included and but won't load if the runtime SDL is older so `sdl_supports_primary_selection` is always true.

Reply:
I don't think I'm getting what you're trying to say.

If you compile with new SDL and use old runtime SDL, `sdl_supports_primary_selection` is false. `SDL_SetPrimarySelectionText` is in this case never loaded (as the code that does the symbol lookup is never reached).

Only if someone enables full relro (or if their OS does things differently), they'll only be able to launch minetest with a new runtime. But this not the default.

Reply:
Huh, I didn't even know this was a thing and thought that all symbols were resolved at startup. I certainly can't remember ever seeing a "unresolved symbol" error at _runtime_ in the 10(?) years I've been using Linux.
Regarding relro: Arch enables full relro, Ubuntu enables relro too (not sure if full), [Fedora has full relro too](https://www.redhat.com/en/blog/hardening-elf-binaries-using-relocation-read-only-relro) so in practice this code is still pretty pointless.

Reply:
Well, yes, this code could probably be left away. But it doesn't hurt.
(And I'm certainly not going to implement a manual dlsym lookup for the destros shipping with full relro. x))
(Btw., FYI it's "symbol lookup error" and "undefined symbol" if it happens at runtime. (Just tried out.))

Reply:
> thought that all symbols were resolved at startup

On musl, that’s always the case. But on glibc, the default is to delay resolve until the first call (or at least that was the default last time I checked).